app/src/main/java/org/mozilla/rocket/content/NewsSourceManager.java

```diff
@@ -67,6 +73,15 @@ public void setNewsSource(String newsSource) {
         ContentRepository.reset();
     }
 
+    public String getNewsSourceUrl() {
+        return newsSourceUrl;
+    }
+
+    public void setNewsSourceUrl(String newsSourceUrl) {
+        this.newsSourceUrl = newsSourceUrl;
+        ContentRepository.resetSubscriptionUrl(newsSourceUrl);
```

Comment:
why not reuse reset()?

Reply:
For this case, it should not set `onDataChangedListener` as null in reset()